src/integTest/kotlin/io/snyk/SnykBulkFileListenerTest.kt

```diff
@@ -101,27 +100,95 @@ class SnykBulkFileListenerTest : BasePlatformTestCase() {
     fun testCurrentIacResults_shouldDropCachedResult_whenIacSupportedFileChanged() {
         val file = "k8s-deployment.yaml"
         val filePath = "/src/$file"
+        createFakeIacResultInCache(file, filePath)
+
+        myFixture.configureByText(file, "some text")
+
+        await().atMost(2, TimeUnit.SECONDS).until { cacheUpdated(filePath) }
+    }
+
+    private fun cacheUpdated(filePath: String?): Boolean {
+        val iacCachedIssues = project.service<SnykToolWindowPanel>().currentIacResult!!.allCliIssues!!
+        return iacCachedIssues.any { iacFile ->
+            (filePath == null || iacFile.targetFilePath == filePath) && iacFile.obsolete
+        }
+    }
+
+    private fun isIacUpdateNeeded(): Boolean =
+        project.service<SnykToolWindowPanel>().currentIacResult?.iacScanNeeded ?: true
+
+    private fun createFakeIacResultInCache(file: String, filePath: String) {
         val iacIssuesForFile = IacIssuesForFile(emptyList(), file, filePath, "npm")
         val iacVulnerabilities = listOf(iacIssuesForFile)
         val fakeIacResult = IacResult(iacVulnerabilities, null)
         val toolWindowPanel = project.service<SnykToolWindowPanel>()
         toolWindowPanel.currentIacResult = fakeIacResult
         val rootIacIssuesTreeNode = toolWindowPanel.getRootIacIssuesTreeNode()
         rootIacIssuesTreeNode.add(IacFileTreeNode(iacIssuesForFile, project))
+    }
 
-        myFixture.configureByText(file, "some text")
+    @Test
+    fun `test current IacResults should mark IacScanNeeded when IaC supported file CREATED`() {
+        val existingFile = "existing.yaml"
+        createFakeIacResultInCache(existingFile, "/src/$existingFile")
+
+        assertFalse(isIacUpdateNeeded())
+
+        myFixture.addFileToProject("new.yaml", "some text")
 
-        await().atMost(2, TimeUnit.SECONDS).until(cacheUpdated(toolWindowPanel, filePath))
+        assertTrue(isIacUpdateNeeded())
+        assertFalse(cacheUpdated(null))
```

Comment:
I dont' understand what is asserted here :(

Reply:
will add message to clarify
otherwise I probably will not understand it either a month later